flash/data/auto_dataset.py

```diff
@@ -0,0 +1,137 @@
+from contextlib import contextmanager
+from copy import deepcopy
+from inspect import signature
+from typing import Any, Callable, Optional, TYPE_CHECKING
+
+import torch
+from pytorch_lightning.core.decorators import parameter_validation
+from pytorch_lightning.trainer.states import RunningStage
+from pytorch_lightning.utilities.warning_utils import rank_zero_warn
+
+from flash.data.process import Preprocess
+from flash.data.utils import _STAGES_PREFIX
+
+if TYPE_CHECKING:
+    from flash.data.data_pipeline import DataPipeline
+
+
+class AutoDataset(torch.utils.data.Dataset):
+
+    FITTING_STAGES = ("train", "val")
+    STAGES = ("train", "test", "val", "predict")
+    DATASET_KEY = "dataset"
+    """
+        This class is used to encapsultate a Preprocess Object ``load_data`` and ``load_sample`` functions.
+        ``load_data`` will be called within the ``__init__`` function of the AutoDataset and ``load_sample``
```

Comment:
```suggestion
        ``load_data`` will be called within the ``__init__`` function of the AutoDataset and ``load_sample``
```
Don't see it being called?

Reply:
Oh, when the `running_stage` is provided. Got it.

Reply:
It is hidden within `self.running_stage = running_stage`.

```
    @running_stage.setter
    def running_stage(self, running_stage):
        if self._running_stage != running_stage or (self._running_stage is None):
            self._running_stage = running_stage
            self._setup(running_stage)
```

```
def _setup(self, stage: RunningStage):
            with self._set_running_stage(stage):
                self._preprocessed_data = self._call_load_data(self.data)
            self._load_data_called = True
```

Reply:
Yeah, figured.